integration/local-gadget/list_containers_test.go

```diff
@@ -143,39 +110,68 @@ spec:
 				c.PodUID = ""
 			}
 
-			var containers []*containercollection.Container
-			if err := json.Unmarshal([]byte(output), &containers); err != nil {
-				return err
-			}
+			return ExpectAllToMatch(output, normalize, expectedContainer)
+		},
+	}
+
+	commands := []*Command{
+		CreateTestNamespaceCommand(ns),
+		PodCommand(cn, "busybox", ns, `["sleep", "inf"]`, ""),
+		WaitUntilPodReadyCommand(ns, cn),
+		listContainersCmd,
+		DeleteTestNamespaceCommand(ns),
+	}
+
+	RunCommands(commands, t)
+}
 
-			if len(containers) == 0 {
-				return fmt.Errorf("expect at least one container in output")
+func TestListWatchContainer(t *testing.T) {
+	t.Parallel()
+	cn := "test-list-watch-container"
+	ns := GenerateTestNamespaceName(cn)
+
+	// TODO: Handle it once we support getting K8s container name for docker
+	// Issue: https://github.com/inspektor-gadget/inspektor-gadget/issues/737
+	if *containerRuntime == ContainerRuntimeDocker {
+		t.Skip("Skip TestListWatchContainer on docker since we don't propagate the Kubernetes pod container name")
+	}
+
+	watchContainersCmd := &Command{
+		Name:         "RunListWatchContainers",
+		Cmd:          fmt.Sprintf("local-gadget list-containers -o json --runtimes=%s --containername=%s --watch", *containerRuntime, cn),
+		StartAndStop: true,
+		ExpectedOutputFn: func(output string) error {
+			expectedContainer := &containercollection.Container{
+				Name:      cn,
+				Podname:   cn,
+				Runtime:   *containerRuntime,
+				Namespace: ns,
 			}
 
-			for _, gotContainer := range containers {
-				normalize(gotContainer)
-				if !reflect.DeepEqual(gotContainer, expectedContainer) {
-					return fmt.Errorf("expect at least one container to match, expected=%+v, got=%+v", expectedContainer, gotContainer)
-				}
+			normalize := func(c *containercollection.Container) {
+				c.ID = ""
+				c.Pid = 0
+				c.OciConfig = nil
+				c.Bundle = ""
+				c.Mntns = 0
+				c.Netns = 0
+				c.CgroupPath = ""
+				c.CgroupID = 0
+				c.CgroupV1 = ""
+				c.CgroupV2 = ""
+				c.Labels = nil
+				c.PodUID = ""
 			}
 
-			return nil
+			return ExpectEntriesToMatch(output, normalize, expectedContainer)
 		},
 	}
 
 	commands := []*Command{
 		CreateTestNamespaceCommand(ns),
-		{
-			Name:           "RunTestListSinglePod",
-			Cmd:            fmt.Sprintf("echo '%s' | kubectl apply -f -", TestPodYaml),
-			ExpectedRegexp: fmt.Sprintf("pod/%s created", po),
-		},
-		{
-			Name:           "WaitForTestListSinglePod",
-			Cmd:            fmt.Sprintf("kubectl wait pod --for condition=ready -n %s %s", ns, po),
-			ExpectedString: fmt.Sprintf("pod/%s condition met\n", po),
-		},
-		listContainersCmd,
+		watchContainersCmd,
+		PodCommand(cn, "busybox", ns, `["sleep", "inf"]`, ""),
```

Comment:
What can we do to ensure `local-gadget` is already running when we start the pod? Maybe a simple `sleep` 😅? I mean, from the output, it is difficult to understand whether `local-gadget` captured it from [the already running containers](https://github.com/inspektor-gadget/inspektor-gadget/blob/5a9c080fc0ac14c63694d91e825b5d6566661a3d/cmd/local-gadget/containers/containers.go#L77-L80), or [the `EventTypeAddContainer` event](https://github.com/inspektor-gadget/inspektor-gadget/blob/5a9c080fc0ac14c63694d91e825b5d6566661a3d/cmd/local-gadget/containers/containers.go#L65-L68). So, we need another way to ensure that, don't you think?

Reply:
We definitely need to ensure this. I hit it couple of times while running tests. For now we do a good old `sleep` and look into other ways latter (if there are any) ?